src/App.tsx

```diff
@@ -3,33 +3,47 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
 
   render() {
+    const { query } = this.state;
+    const visibleMovies = moviesFromServer.filter((movie) => {
+      return movie.title.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+      || movie.description.toLocaleLowerCase().includes(query.toLocaleLowerCase());
+    });
```

Comment:
you can calculate it here, but create separate method and call it here to make code more clear